File: contact/src/main/java/com/example/contact/MainActivity.java
package com.example.contact;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements NewAdapter.SelectedContact {

    private EditText search_editText;
    private RecyclerView recyclerview;
    private ImageButton addButton;
    private NewAdapter adapter;
    private static final int REQUEST_CODE_ADDCONTACT = 1;
    private static final int REQUEST_CODE_REMOVE = 2;
    private static final String KEY_NUMBER = "number";
    private Boolean ff;
    private List<NewAdapter.Contact> newContactList = new ArrayList<>();

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        addButton = findViewById(R.id.addButton);

        recyclerview = findViewById(R.id.recyclerview);
        recyclerview.setAdapter(new NewAdapter(newContactList, this));
        recyclerview.setLayoutManager(new LinearLayoutManager(MainActivity.this, RecyclerView.VERTICAL, false));

        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, AddActivity.class);
                startActivityForResult(intent, REQUEST_CODE_ADDCONTACT);
            }
        });
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == RESULT_OK) {
            switch (requestCode) {
                case REQUEST_CODE_ADDCONTACT:
                    assert data != null;
                    boolean radioButtom_Phone = data.getBooleanExtra("radioButtom_Phone", false);
                    boolean radioButtom_Email = data.getBooleanExtra("radioButtom_Email", false);
                    if (radioButtom_Email) {
                        ff = true;
                    }
                    if (radioButtom_Phone) {
                        ff = false;
                    }
                    String newName = data.getStringExtra("name");
                    String newNumberAndEmail = data.getStringExtra("number_email");
                    if (!newName.trim().isEmpty() && !newNumberAndEmail.trim().isEmpty() && recyclerview.getAdapter() != null) {
                        adapter = (NewAdapter) recyclerview.getAdapter();
                        adapter.addItems(new NewAdapter.Contact(newName, newNumberAndEmail, ff));
                    }
                    break;
                case REQUEST_CODE_REMOVE:
                    String nameRemove = data.getStringExtra("name_remove");
                    String NumberAndEmailREmove = data.getStringExtra("editText_phoneNumber");
                    Boolean flag = data.getBooleanExtra("flag" , false);
                    if (nameRemove.trim().isEmpty() && NumberAndEmailREmove.trim().isEmpty()) {
                        adapter.remove();
                    }
                    if (!nameRemove.trim().isEmpty() && !NumberAndEmailREmove.trim().isEmpty()) {
                        NewAdapter.Contact contact = new NewAdapter.Contact(nameRemove , NumberAndEmailREmove , flag);
                        adapter.upDate(contact);
                    }
                default:
                    break;
            }
        }
    }
    @Override
    public void selectedContact(NewAdapter.Contact contact) {
        Intent intent = new Intent(MainActivity.this, RemoveActivity.class);
        intent.putExtra("mNameList", contact.getNameText());
        intent.putExtra("mNumberList", contact.getNumberText());
        intent.putExtra("flag", contact.getType());
        startActivityForResult(intent, REQUEST_CODE_REMOVE);
    }
}

File: contactkotlin/src/main/java/com/example/contactkotlin/MainActivity.kt
package com.example.contactkotlin

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import android.view.View
import android.widget.EditText
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.contactkotlin.ItemAdapter.Contact
import java.util.*


class MainActivity() : AppCompatActivity(),  ItemAdapter.SelectedContact {

    private lateinit var search_editText: EditText
    private lateinit var recyclerview: RecyclerView
    private lateinit var addButton: ImageButton
    private lateinit var adapter: ItemAdapter
    private var ff: Boolean = false
    private val newContactList: MutableList<ItemAdapter.Contact> = ArrayList<ItemAdapter.Contact>()

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        addButton = findViewById(R.id.addButton)
        recyclerview = findViewById(R.id.recyclerview)
        recyclerview.setAdapter(ItemAdapter(newContactList,this))
        recyclerview.setLayoutManager(
                LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        )
        addButton.setOnClickListener(View.OnClickListener { val intent = Intent(this, AddActivity::class.java)
            startActivityForResult(intent, REQUEST_CODE_ADDCONTACT)
        })
    }

    public override fun onActivityResult(
            requestCode: Int,
            resultCode: Int,
            data: Intent?
    ) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK) {
            when (requestCode) {
                REQUEST_CODE_ADDCONTACT -> {
                    val radioButtom_Phone = data!!.getBooleanExtra("radioButtom_Phone", false)
                    val radioButtom_Email = data!!.getBooleanExtra("radioButtom_Email", false)
                    if (radioButtom_Email) {
                        ff = true
                    }
                    if (radioButtom_Phone) {
                        ff = false
                    }
                    val newName = data!!.getStringExtra("name")
                    val newNumberAndEmail = data!!.getStringExtra("number_email")
                    if (!newName.trim { it <= ' ' }.isEmpty() && !newNumberAndEmail.trim { it <= ' ' }.isEmpty() && recyclerview.adapter != null) {
                        adapter = recyclerview.adapter as ItemAdapter
                        adapter.addItems(Contact(newName, newNumberAndEmail, ff))
                    }
                }
                REQUEST_CODE_REMOVE -> {
                    val nameRemove = data!!.getStringExtra("name_remove")
                    val NumberAndEmailREmove = data!!.getStringExtra("editText_phoneNumber")
                    val flag = data!!.getBooleanExtra("flag", false)
                    if (nameRemove.trim { it <= ' ' }.isEmpty() && NumberAndEmailREmove.trim { it <= ' ' }.isEmpty()) {
                        adapter.remove()
                    }
                    if (!nameRemove.trim { it <= ' ' }.isEmpty() && !NumberAndEmailREmove.trim { it <= ' ' }.isEmpty()) {
                        val contact = Contact(nameRemove, NumberAndEmailREmove, flag)
                        adapter.upDate(contact)
                    }
                }
                else -> {
                }
            }
        }
    }

    override fun selectedContact(contact: Contact) {
        val intent = Intent(this, RemoveActivity::class.java)
        intent.putExtra("mNameList", contact.nameText)
        intent.putExtra("mNumberList", contact.numberText)
        intent.putExtra("flag", contact.type)
        startActivityForResult(intent, REQUEST_CODE_REMOVE)
    }

    companion object {
        private const val REQUEST_CODE_ADDCONTACT = 1
        private const val REQUEST_CODE_REMOVE = 2
        private const val KEY_NUMBER = "savedInstanceState"
    }
}
